### PR TITLE
WIP: Rewriting various simpleserialize functionality to accomodate to the new version of the @chainsafe/ssz lib

### PR DESCRIPTION
**Pull request by 3xtr4t3rr3str14l**

### CODE REVIEW

File: src/util/random.ts [hunk 1]
```diff
@@ -1,61 +1,66 @@
...
+    return randomNumberUint();
+  }
+  else if(isBigIntUintType(type)) {
+    return randomBigUint();
+  }
+  else if(isBooleanType(type)) {
+    return randomBoolean();
+  }
+  else if(isBitVectorType(type)) {
+    return randomBooleanArray(64);
```
Comment (wemeetagain) [inline src/util/random.ts:40]: 64 => type.length

File: src/util/random.ts [hunk 2]
```diff
@@ -1,61 +1,66 @@
...
-      return Array.from({length: type.length}, () => _createRandomValue(type.elementType));
-    case Type.container:
-      const obj: any = {};
-      type.fields.forEach(([fieldName, fieldType]) => {
-        obj[fieldName] = _createRandomValue(fieldType);
-      });
-      return obj;
+export function createRandomValue(type: BasicType<unknown> | CompositeType<object>): any {
+  if(isNumberUintType(type)) {
+    return randomNumberUint();
```
Comment (wemeetagain) [inline src/util/random.ts:31]: should pass the `type.byteLength` or similar. Eg: Uint8 has a range of 0-255

File: src/util/random.ts [hunk 3]
```diff
@@ -1,61 +1,66 @@
...
+  }
+  else if(isBigIntUintType(type)) {
+    return randomBigUint();
```
Comment (wemeetagain) [inline src/util/random.ts:34]: Same as number uint

File: src/util/random.ts [hunk 4]
```diff
@@ -1,61 +1,66 @@
...
+    return randomBoolean();
+  }
+  else if(isBitVectorType(type)) {
+    return randomBooleanArray(64);
+  }
+  else if(isByteVectorType(type)) {
+    return randomByteVector(type.length);
+  }
+  else if(isBitListType(type)) {
+    const randomLength = Math.floor(Math.random() * 512) + 16;
```
Comment (wemeetagain) [inline src/util/random.ts:46]: This should be Math.min with type.limit, as done with ListType below

File: src/util/random.ts [hunk 5]
```diff
@@ -1,61 +1,66 @@
...
+  }
+  else if(isContainerType(type)) {
+    const obj: any = {};
+    Object.entries(type.fields).forEach(([fieldName, fieldType]) => {
+      obj[fieldName] = createRandomValue(fieldType);
+    });
+    return obj;
+  }
+  else {
+    return 'N/A';
```
Comment (wemeetagain) [inline src/util/random.ts:64]: throw an error here, this should be an exhaustive list

File: src/util/input_types.ts [hunk 6]
```diff
@@ -1,25 +1,24 @@
...
+    parse: <T>(raw: string, type: Type<T>) => any,
+    dump: <T>(value: any, type: Type<T>) => string,
 }
 
 export const inputTypes: InputTypeRecord = {
     yaml: {
-        parse: (raw, type) => expandInput(parseYaml(raw), parseType(type), false),
-        dump: (value, type) => dumpYaml(unexpandInput(value, parseType(type), false)),
+        parse: (raw, type) => type.fromJson(parseYaml(raw)),
+        dump: (value, type) => dumpYaml(type.toJson(typeof value === 'number' ? value.toString() : value)),
```
Comment (wemeetagain) [inline src/util/input_types.ts:14]: What is the `typeof` check for?

Comment (3xtr4t3rr3str14l) [reply src/util/input_types.ts:14]: that's when we had talked about trying to convert numbers to strings for this part

File: package.json [hunk 7]
```diff
@@ -13,9 +13,9 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.5.5",
-    "@chainsafe/eth2.0-config": "^0.1.1",
-    "@chainsafe/eth2.0-types": "^0.1.1",
-    "@chainsafe/ssz": "0.5.2",
+    "@chainsafe/lodestar-config": "^0.5.0",
+    "@chainsafe/lodestar-types": "^0.5.0",
```
Comment (wemeetagain) [inline package.json:17]: you can update the lodestar packages to 0.6.0

File: package.json [hunk 8]
```diff
@@ -13,9 +13,9 @@
...
+    "@chainsafe/ssz": "^0.6.2",
```
Comment (wemeetagain) [inline package.json:18]: and ssz 0.6.3

File: package.json (hunk 7 above)
Comment (3xtr4t3rr3str14l) [reply package.json]: updated to 0.7.0

File: package.json (hunk 8 above)
Comment (3xtr4t3rr3str14l) [reply package.json]: updated to 0.6.4

**Comment by 3xtr4t3rr3str14l** [comment]
> Looking good! can you JSON stringify with `JSON.stringify(data, null, 2)` to "pretty-print" it?

@wemeetagain  done

**Pull request merged**